ddtrace/tracer/open_spans.go

```diff
@@ -0,0 +1,180 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package tracer
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+type BLList[T comparable] struct {
+	head *listNode[*LList[T]]
+	tail *listNode[*LList[T]]
+}
```

Comment:
I wonder if generics are actually needed here. Using generics can have performance implications, and makes things more complex, so if we don't fully need it, then we should probably avoid it. It also makes the code a bit harder to read, since things are just called `Element` instead of "span" or "trace", which have more meaning. Can you explain why generics is needed here?

Instead, it might work to just have a linked list that looks like this:

```
type TraceList struct {
  head *spanList
}

type spanList struct {
  head *spanNode
}

type spanNode struct {
  spanId, traceId string
  start time.Time
}
```

Or instead of using a `spanNode` you could just use a `*span`. The benefit of using a custom type is that it may make sorting by `start` easier/faster. The benefit of using `span` we already have is that the memory has already been allocated, so we can just keep another reference to the pointer, which shouldn't cost much. Depends what we're optimizing for.

Reply:
I was planning to use generics here to allow for storing different values/types (i.e. traces vs spans), but I definitely think that it's not as useful anymore. Removing the generics might also help w cleaning up the code a bit. I can do that!